backend/test_observer/controllers/test_results/test_results.py

```diff
@@ -0,0 +1,307 @@
+        except ValueError:
+            filters.append(Artefact.family.is_(None))
+
+    if environment:
+        filters.append(Environment.name.ilike(f"%{environment}%"))
+
+    if test_case:
+        filters.append(TestCase.name.ilike(f"%{test_case}%"))
+
+    if template_id:
+        filters.append(TestCase.template_id == template_id)
+
+    if issues:
+        # Handle multiple issue IDs separated by commas
+        issue_list = [issue.strip() for issue in issues.split(",") if issue.strip()]
+        if issue_list:
+            try:
+                issue_ids = [int(issue_id) for issue_id in issue_list]
+                filters.append(
+                    TestResult.id.in_(
+                        select(IssueTestResultAttachment.test_result_id)
+                        .join(IssueTestResultAttachment.issue)
+                        .where(Issue.id.in_(issue_ids))
+                    )
+                )
+            except ValueError:
+                return TestResultSearchResponse(count=0, test_results=[])
+
+    if from_date:
+        filters.append(TestExecution.created_at >= from_date)
+
+    if until:
+        filters.append(TestExecution.created_at <= until)
+
+    # Build the main query with window function for total count in a single query
+    query = (
+        select(TestResult, func.count().over().label("total_count"))
+        .join(TestResult.test_execution)
+        .join(TestExecution.environment)
+        .join(TestExecution.artefact_build)
+        .join(ArtefactBuild.artefact)
+        .join(TestResult.test_case)
+        .options(
+            joinedload(TestResult.test_execution).joinedload(TestExecution.environment),
+            joinedload(TestResult.test_execution)
+            .joinedload(TestExecution.artefact_build)
+            .joinedload(ArtefactBuild.artefact),
+            joinedload(TestResult.test_case),
+        )
+        .order_by(desc(TestExecution.created_at), desc(TestResult.id))
+        .offset(offset)
+        .limit(limit)
+    )
+
+    # Apply all filters
+    if filters:
+        query = query.where(and_(*filters))
+
+    result = db.execute(query).all()
+
+    if result:
+        test_results = [row[0] for row in result]
+        total_count = result[0][1]
+    else:
+        test_results = []
+        total_count = 0
+
+    return TestResultSearchResponse(count=total_count, test_results=test_results)
+
+
+@router.get("/families", response_model=list[str])
+def get_families(db: Session = Depends(get_db)) -> list[str]:
+    """
+    Returns all families that have test results.
+    """
+    families = (
+        db.execute(
+            select(distinct(Artefact.family))
+            .join(ArtefactBuild)
+            .join(TestExecution)
+            .join(TestResult)
+            .order_by(Artefact.family)
+        )
+        .scalars()
+        .all()
+    )
+
+    return [family.value for family in families]
+
+
+@router.get("/environments", response_model=list[str])
```

Comment:
It's a bit annoying but I think this route should be `GET /v1/environments`, meaning it'll probably need a new `test_observer/controllers/environments` folder.

Looking at it here, that may make sense to put in it's own PR/ticket since `GET /v1/test_results` can be implemented without it.